File: matlab/incremental.m
clear
clear all
clc

func = input('Ingrese la funcion: ', 's');
f = inline(func);
x0 = input('Ingrese el valor inicial: ');
delta = input('Ingrese el delta: ');
iter = input('Ingrese el numero de iteraciones');
y0 = f(x0);

fprintf('\n |   x  |  f(x)   |');
fprintf('\n ------------------');

fprintf('\n |   %g   |  %g  |', x0,y0);

if y0 == 0
	fprintf('\n %g es una raiz',x0);
else
	if delta <= 0
		fprintf('\n Ingrese un delta valido');
	else
        if  iter <= 0
            fprintf('\n Ingrese un número de iteraciones valido');
        else
            contador = 1;
            x1 = x0 + delta;
            y1 = f(x1);
            while y0*y1 > 0 && contador < iter
              x0 = x1;
              y0 = y1;
              fprintf('\n |   %g   |  %g  |', x0,y0);
              x1 = x0 + delta;
              y1 = f(x1);
              contador = contador + 1;
            end
            fprintf('\n |   %g   |  %g  |', x1,y1);
            if y0*y1 < 0
              fprintf('\n [%g,%g] es un intervalo que contiene una raiz.', x0,x1);
            elseif y1 == 0
              fprintf('\n %g es una raiz.', x1);
            else
              fprintf('\n Fracaso en %g iteraciones.', iter);
            endif

        endif
    fprintf('\n')
    ezplot(f)
	endif
endif

File: Flutter/analisis_numerico/lib/Backend/incremental.dart
import 'dart:collection';

import 'package:analisis_numerico/Commons/funciones.dart';

class IncrementalAlgorithm{

  Map tabla = new Map();
  Map get getTabla => tabla;

  String evaluateAlgorithm({double x0, int iteraciones, double delta}){
    double y0 = f(x0);
    if(y0 == 0){
      return "$x0 es una raiz";
    }else if(delta == 0){
      return "Delta inapropiado";
    }else if(iteraciones <= 0){
      return "Valor de iteraciones inapropiado";
    }else{
      double x1, y1;
      int contador = 1;
      x1 = x0 + delta;
      y1 = f(x1);
      tabla[contador]=[x0,y0];
      while(y0*y1 > 0 && contador < iteraciones){
        x0 = x1;
        y0 = y1;
        x1 = x0 + delta;
        y1 = f(x1);
        contador++;
        tabla[contador]=[x0,y0];
      }
      if(y0*y1 < 0){
        tabla[contador]=[x0,y0];
        tabla[contador+1]=[x1,y1];
        return "[$x0,$x1] definen un intervalo";
      }else if(y1 == 0){
        return "$x1 es una raiz";
      }else{
        return "Fracaso en $iteraciones iteraciones";
      }
    }
  }
}
